packages/gatsby/cache-dir/navigation.js

```diff
@@ -18,9 +18,8 @@ function maybeRedirect(pathname) {
 
   if (redirect != null) {
     if (process.env.NODE_ENV !== `production`) {
-      const pageResources = loader.loadPageSync(pathname)
-
-      if (pageResources != null) {
+      // If such page exist show the error. If not it was intentional.
+      if (!loader.isPageNotFound(pathname)) {
```

Comment:
I did run few tests with this and started to think that replacing one with the other might not actually be the way to go and we might need both checks.

`loader.isPageNotFound` will be false if user doesn't have 404 page and there is a page for given pathname. So this means that now users without 404 page will get those error messages. This also makes it tricky to test, as we would have to do another build for no-404 case

Doing a little matrix for this:

|                       | Path exists + have 404 | Path exists + doesn't have 404 | Path doesn't exist + have 404 | Path doesn't exist + doesn't have 404 |
|-----------------------|------------------------|--------------------------------|-------------------------------|---------------------------------------|
| loader.isPageNotFound | false                  | false                          | true                          | false                                 |
| loader.loadPageSync   | { ...data }            | { ...data }                    | { ... data } (for 404 page)   | undefined                             |

We want to show error (probably should have been warning) message when both `loader.isPageNotFound(X) === false && loader.loadPageSync(X) !== null`?

Reply:
Maybe some other method in loader - like `pageExists(pathname)` would be better, as our navigation code shouldn't really need to know that much about loader stuff

Reply:
@pieh I am totally agree with you to show warning message instead of error. pageExisits method sounds amazing. Will work on creating that method.

Reply:
@mnurik any update on this from your? 😃

Reply:
@blainekasten sorry, long vacation :D Working on solution

Reply:
@blainekasten sorry, long vacation :D Working on solution

Reply:
ok, we got back here after other fixes that happened not in this PR, and right now this is actually working as intended